src/averageAges.js

```diff
@@ -36,8 +39,16 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((woman) => woman.sex === 'f'
+    && (withChildren === undefined
+    || people.find((motherName) => motherName.mother === woman.name)));
+
+  const averageWomensAge = Math.round(women.reduce((sum, currentPerson) => sum
+    + currentPerson.died - currentPerson.born, 0) / women.length * 100) / 100;
```

Comment:
We have the same callback for men to calculate age, let's create a separate function.